src/main/java/Duke.java

```diff
@@ -1,10 +1,227 @@
+            taskList.add(new Deadline(task, by));
+            printTaskAssignment(taskList.size() - 1);
+
+        } else if (text.startsWith("event")) {
+            //Assign event task
+            task = text.substring(6, text.indexOf(" \\at"));
+            at = text.substring(text.indexOf("\\at") + 4, text.length());
+            taskList.add(new Event(task, at));
+            printTaskAssignment(taskList.size() - 1);
+
+        } else if (text.startsWith("todo")) {
+            //Assign todos task
+            task = text.replace("todo", "");
+            if (task.length() < 1) {
+                throw new StringIndexOutOfBoundsException();
+            }
+            taskList.add(new Todo(task));
+            printTaskAssignment(taskList.size() - 1);
+
+        } else {
+            throw new DukeException();
+        }
+    }
+
+    public static void displayList() {
+        int index = 1;
+        System.out.println("\t____________________________________________________________");
+        System.out.println("\t Here are the tasks in your list:");
+        for (Task listTask : taskList) {
+            if (listTask == null) {
+                break;
+            }
+            System.out.println("\t " + index + "." + "[" + listTask.getTaskSymbol() + "]" + "["
+                    + listTask.getStatusIcon() + "] " + listTask);
+            index++;
+        }
+        System.out.println("\t____________________________________________________________");
+    }
+
+    public static void printTaskAssignment(int index) {
+        int taskIndex = index + 1;
+        System.out.println("\t____________________________________________________________");
+        System.out.println("\t Got it. I've added this task:");
+        System.out.println("\t\t" + "[" + taskList.get(index).getTaskSymbol() + "]" + "[" +
+                taskList.get(index).getStatusIcon() + "]" + " " + taskList.get(index));
+        System.out.println("\t Now you have " + taskIndex + " tasks in the list.");
+        System.out.println("\t____________________________________________________________");
+    }
+
+/*    private static void printFileContents(String filePath) throws FileNotFoundException {
+        File f = new File(FILE_PATH);
+        Scanner s = new Scanner(f);
+        while (s.hasNext()) {
+            System.out.println(s.nextLine());
+        }
+    }*/
+
+    public static void loadProgress() throws FileNotFoundException {
+        //File contains something
+        String textInput;
+        Scanner s = new Scanner(f);
+        String[] taskArray = new String[4];
+        //Format of text: T | 1 | read book
+        while (s.hasNext()) {
+            textInput = s.nextLine();
+            taskArray = textInput.split(" \\| ");
+            if (taskArray[0].equalsIgnoreCase("T")) {
+                //Create Todo task
+                taskList.add(new Todo(taskArray[2]));
+                taskList.get(taskCount).setLoadStatus(taskArray[1]);
```

Comment:
I am having difficulties to find the initialization for the "taskCount" variable. Could you help me out with that?

Reply:
taskCount variable is a static variable declared in Duke class line 9

Reply:
Yes, I have found that. But you left it without initialization. But it would be better you have a initilization of it value when you load tasks.